include/tgfx/core/Matrix.h

```diff
@@ -159,8 +182,30 @@ class Matrix {
    * @return  Returns true if the Matrix has no effect.
    */
   bool isIdentity() const {
-    return values[0] == 1 && values[1] == 0 && values[2] == 0 && values[3] == 0 && values[4] == 1 &&
-           values[5] == 0;
+    return this->getType() == 0;
+  }
+
+  /**
+   * Returns true if Matrix at most scales and translates. Matrix may be identity,
+   * contain only scale elements, only translate elements, or both. Matrix form is:
+   *        | scale-x    0    translate-x |
+   *        |    0    scale-y translate-y |
+   *        |    0       0         1      |
+   * @return  true if Matrix is identity; or scales, translates, or both
```

Comment:
Returns true if the Matrix only performs scaling and translation, or is the identity.
The Matrix may be an identity matrix, contain only scale elements, only translation elements, or both.
Matrix form:
        | scale-x    0    translate-x |
        |   0     scale-y translate-y |
        |   0        0         1      |